stories/Navigation/Navigation.stories.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { Meta, Story } from '@storybook/react';
+import { AppleAlt } from '@styled-icons/fa-solid/AppleAlt';
+import { Archive } from '@styled-icons/fa-solid/Archive';
+import { BrowserRouter } from 'react-router-dom';
+import { Compass } from '@styled-icons/fa-solid/Compass';
+import { Navigation, NavigationProps, NavigationFooter } from '../../src';
+import { createStoryTitle } from '../Constants';
+import { logoWhite } from '../assets';
+
+export default {
+    title: createStoryTitle('Navigation'),
+    component: Navigation,
+} as Meta;
+
+const Header = () => <div>hello</div>;
+
+const defaultArgs = {
+    label: 'Some Text', // this is for NavigationHeader
+};
+
+const Template: Story<NavigationProps> = (args) => (
+    <BrowserRouter>
+        <Navigation {...args} />
+    </BrowserRouter>
+);
+
+export const Basic = Template.bind({});
+
+Basic.args = {
+    ...defaultArgs,
+};
+
+export const WithPages = Template.bind({});
+
+WithPages.args = {
+    ...defaultArgs,
+    pages: {
+        Item1: {
+            navProps: {
+                to: '/:id/orders',
+                icon: AppleAlt,
+            },
+        },
+        Item2: {
+            navProps: {
+                to: '/:id/orders',
+                icon: Archive,
+            },
+        },
+    },
+};
+
+export const WithFooter = Template.bind({});
+
+WithFooter.args = {
+    ...defaultArgs,
+    footer: <NavigationFooter url="#foo" text="Link to" icon={Compass} />,
+};
+
+export const WithHeader = Template.bind({});
+WithHeader.args = {
+    ...defaultArgs,
+    header: <Header />,
```

Comment:
This should be NavigationHeader? Display is off with Header.

Reply:
No. this is like a sub header. It is optional. NavigationHeader it is used inside Navigation Component.

Reply:
Alright, so same as the bumped comments, we want to make components in the React-UI library extensible. This means that they can be used for many purposes not just one. The way to make this extensible would be to pass in NavigationHeader component and display that rather than hard coding NavigationHeader into the component.

This way if someone wants to add an element underneath navigationHeader or above it and style they are not locked into having the NavigationHeader in that location. They can create the Header for the use case.

I know for this specific instance of Navigation this will likely be the layout. However, part of adding components to the library is to make them usable in many use cases.